modular_zubbers/code/modules/wizard_dize/wizard_dice_object.dm

```diff
@@ -0,0 +1,199 @@
+
+	switch(smite_rng_seed % 9)
+		if(0)
+			//Bad luck.
+			target.AddComponent(/datum/component/omen/smite, incidents_left = 7) //7 (years) bad luck
+			to_chat(target, span_warning("You get a bad feeling about this..."))
+		if(1)
+			//Drain bamage.
+			target.adjustOrganLoss(ORGAN_SLOT_BRAIN, BRAIN_DAMAGE_DEATH - 1, BRAIN_DAMAGE_DEATH - 1)
+			to_chat(target, span_warning("You feel <b>stupid</b> about rolling [src]..."))
+		if(2)
+			//Forced to speak a random language.
+			target.apply_status_effect(/datum/status_effect/tower_of_babel, teleport_delay_pickup)
+			to_chat(target, span_warning("Come, let us go down and confuse their language there, so that they will not understand one another's speech. It'd be fucking hilarious!"))
+		if(3)
+			//Raining fireball.
+			target.Immobilize(3 SECONDS)
+			new /obj/effect/temp_visual/target(get_turf(target))
+			to_chat(target, span_warning("I CAST FIREBALL."))
+		if(4)
+			//Lighting bolt smite.
+			var/turf/lightning_source = get_turf(src)
+			lightning_source.Beam(target, icon_state="lightning[rand(1,12)]", time = 5)
+			target.adjustFireLoss(LIGHTNING_BOLT_DAMAGE)
+			playsound(get_turf(target), 'sound/magic/lightningbolt.ogg', 50, TRUE)
+			target.electrocution_animation(LIGHTNING_BOLT_ELECTROCUTION_ANIMATION_LENGTH)
+			to_chat(target, span_warning("LIGHTNING BOLT!!"))
+		if(5)
+			//Ick Ock Phobia
+			target.gain_trauma(/datum/brain_trauma/mild/phobia/ocky_icky, TRAUMA_RESILIENCE_SURGERY)
+			to_chat(target, span_warning("Those damn coders are out to get us!!!"))
+		if(6)
+			//A rod comes and fucks you up. And the station.
+			var/turf/target_turf = get_turf(target)
+			var/startside = pick(GLOB.cardinals)
+			var/turf/start_turf = spaceDebrisStartLoc(startside, target_turf.z)
+			var/turf/end_turf = spaceDebrisFinishLoc(startside, target_turf.z)
+			new /obj/effect/immovablerod(start_turf, end_turf, target, FALSE)
+			to_chat(target, span_warning("Huh. Nothing seems to have happened. I guess I am unstoppable..."))
+		if(7)
+			//A pod comes down and slams you.
+			target.Immobilize(3 SECONDS)
+			podspawn(list(
+				"target" = get_turf(target),
+				"path" = /obj/structure/closet/supplypod/centcompod,
+				"style" = STYLE_CENTCOM,
+				"spawn" = /obj/item/toy/plush/lizard_plushie,
+				"damage" = 50,
+				"explosionSize" = list(0, 0, 2, 4),
+				"effectStun" = TRUE
+			))
+			to_chat(target, span_warning("SPECIAL DELIVERY!"))
+		if(8)
+			//You turn into a lizard. If you're already a lizard, you're now scared of lizards.
+			if(target.dna && !istype(target.dna.species,/datum/species/lizard/))
+				target.set_species(/datum/species/lizard/, TRUE, FALSE, null, null, null, null, TRUE, TRUE)
+				to_chat(target, span_warning("You're a lizard, Harry."))
+			else
+				target.gain_trauma(/datum/brain_trauma/mild/phobia/lizards, TRAUMA_RESILIENCE_SURGERY)
+				to_chat(target, span_warning("Wait a minute, I've been a LIZARD THIS WHOLE TIME?"))
+
+
+
+	smite_rng_seed++
```

Comment:
Doesnt this make the effects completely predictable?

Reply:
Yes, but that should legitimately be a non-issue unless you've somehow managed to get every die of fate roll.